demos/nprk/demo_burgers_nprk.py.rst

```diff
@@ -0,0 +1,106 @@
+
+.. math::
+
+   \partial_t u + u\, \partial_x u - \epsilon\, \partial_x^2 u = 0
+
+To demonstrate the non-linearly partitioned RK (NPRK) method, various partitionings are listed and compared in:
+
+Buvoli, Southworth "A New Class of Runge-Kutta Methods for Nonlinearly Partitioned Systems", arXiv 2024
+
+The non-linear partitioning is explained here and implemented in Irksome.
+
+A non-linear partitioning equates the time derivative with an operator taking two arguments, :math:`u_t=F(u,u)`.
+Each stage of the NPRK scheme treats one argument implicitly and the other explicitly.
+There are many choices of non-linear partitioning, but
+
+.. math::
+
+   F(u_1, u_2) = - u_2\, \partial_x u_1 + \epsilon\, \partial_x^2 u_1
+
+is selected.
+
+With homogenous Dirichlet boundary conditions, this equation gives rise to the weak form
+
+.. math::
+
+   (v, \partial_t u) + (v, u_2\, \partial_x u_1) + (\partial_x v, \epsilon\, \partial_x u_2) = 0
+
+Following the paper, this equation is solved numerically for :math:`\epsilon=1/200`, :math:`x\in[-2,2]`, :math:`t\in[0,0.6]`
+with :math:`1000` grid points and the initial condition
+
+.. math::
+
+   u(x,0)=e^{-3x^2}
+
+The timestep :math:`\Delta t=6\cdot 10^{-3}` is selected as the middle of the range tested in the referenced paper.
+
+To implement this, begin by importing firedrake::
+
+   from firedrake import *
+
+Additionally, a few basic items and the NPRK time stepper are imported from irksome::
+
+   from irksome import Dt, MeshConstant, NPRKMethod
+
+The NPRK tableaux are implemented in :class:`irksome.nprk_tableaux.NPRKTableau`.
+If each argument is treated implicitly on at least one stage, the tableau is considered implicit-implicit (IMIM).
+If only one argument is treated implicitly across all stages, the tableau is implicit-explicit (IMEX).
+IMIM tableaux can be accessed through :class:`irksome.IMIMNPRKTableau` and IMEX through :class:`irksome.IMEXNPRKTableau`.
+For this problem, the NPRK-IMEX forward Euler tableau is selected::
+
+   from irksome import IMEXNPRKTableau
+
+   nprk_tableau = IMEXNPRKTableau("euler")
+
+The space of piecewise linear, continuous functions is chosen and initialized alongside the mesh in the standard Firedrake approach::
+
+   N = 1000
+   x0 = -2.0
+   x1 = 2.0
+
+   msh = IntervalMesh(N, x0, x1)
+   V = FunctionSpace(msh, "CG", 1)
+
+Variables for the current time step and time value, as well as the simulation time, are defined by::
+
+   MC = MeshConstant(msh)
+   dt = MC.Constant(6e-3)
+   t = MC.Constant(0.0)
+
+   TMAX = 0.6
+
+The initial condition is defined and assigned via interpolation::
+
+   x = SpatialCoordinate(msh)
+
+   u0 = exp(-3*x**2)
+
+   u = Function(V)
+   u.interpolate(u0)
+
+To define the weak form, two additional :class:`firedrake.Function`'s are needed to indicate :math:`u_1` and :math:`u_2` ::
+
+   u_1, u_2 = Function(V), Function(V)
+   EPS = Constant(1/200)
+   v = TestFunction(V)
+
+   F = inner(v, Dt(u))*dx + inner(v, u_2*u_1.dx(0))*dx \
```

Comment:
I'm concerned here about the interface relative to the additive/IMEX approach.  This has the original function `u` plus two extra functions (somehow linked) that play roles in the splitting.  I don't see a better approach short of labeling (and that might be hard multiplicatively within a single integral?)